embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/orchestrator/SessionOrchestratorImpl.kt

```diff
@@ -38,11 +42,13 @@ internal class SessionOrchestratorImpl(
 
     override fun onForeground(coldStart: Boolean, timestamp: Long) {
         backgroundActivityService?.endBackgroundActivityWithState(timestamp)
```

Comment:
All these methods to start/stop sessions are running serially because they are always done on a particular thread?

Reply:
Just keeping existing behavior for now, which is to run on whatever the calling thread is. `SessionService` does use a lock that I want to move up to this class eventually.